### PR TITLE
Rollback any document changes if adding a nuget package fails.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes: https://github.com/dotnet/roslyn/issues/8691

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
test this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
retest this please

**Comment by CyrusNajmabadi** [comment]
test this please

**Comment by CyrusNajmabadi** [comment]
retest this please

File: src/Features/Core/Portable/AddImport/CodeActions/PackageReference.ParentCodeAction.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                    SyntaxNode node,
+                    bool placeSystemNamespaceFirst,
+                    string versionOpt,
+                    bool isLocal)
+                {
+                    var title = versionOpt == null
+                        ? FeaturesResources.Find_and_install_latest_version
+                        : isLocal
+                            ? string.Format(FeaturesResources.Use_local_version_0, versionOpt)
+                            : string.Format(FeaturesResources.Install_version_0, versionOpt);
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/CodeActions/PackageReference.ParentCodeAction.cs]: Where does this preferred version come from?  Should it be first in the list?

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
test this please

**Comment by CyrusNajmabadi** [comment]
test this please

**Pull request merged**